django/contrib/admin/widgets.py

```diff
@@ -194,7 +195,23 @@ def label_for_value(self, value):
         key = self.rel.get_related_field().name
         try:
             obj = self.rel.model._default_manager.using(self.db).get(**{key: value})
-            return '&nbsp;<strong>%s</strong>' % escape(Truncator(obj).words(14, truncate='...'))
+            change_url = reverse(
+                '%s:%s_%s_change' % (
+                    self.admin_site.name,
+                    obj._meta.app_label,
+                    obj._meta.object_name.lower(),
+                ),
+                args=(obj.pk,)
+            )
+            return '&nbsp;<strong><a href="%s">%s</a></strong>' % (
```

Comment:
The `else` would be redundant here; it could just go after the `try/except` block (since the except has a `return` anyway).  
Or is this against coding conventions?